### ISSUE TITLE
Smi, Value32/64

### ISSUE THREAD

**Issue by zherczeg** [Samsung/escargot#87]
I am thinking about working on an optimization, but before I do anything, I would like to ask your opinion about it. As far as I see Escargot uses three different representation for values: an SmallValue, which contains `uintptr_t` data, and a Value which size is always 8 byte, but there are two different implementations for 32 and 64 bit systems. 

The problem is conversion costs: Value->SmallValue and SmallValue->Value happens frequently. It would be good to reduce these conversion costs. I am thinking about using common constants for Value and SmallValue for primitive types (true/false,undefined). Do you think this is possible?

In general, is it worth to have these two representations? If we would keep only one, which one would be better: SmallValue or Value?

(As far as I know returning with two uint32 on 32 bit machines is efficient since it just use two registers)

**Comment by ksh8281** [comment]
A. we tested large Value array typed malloc, but performance is too bad..(versus SmallValue array)
because telling to bdwgc element of array are POD or pointer is slow.
but even internal value of SmallValue is always pointer, odd internal value of SmallValue is always integer, 
so we don't have to tell to bdwgc element of array is POD or pointer

B. we tested doing common operations with SmallValue. but it is slow...
testing it's type on every operation takes much time. because there is no type information(int, undefined, null, pointer-value...) saved directly(versus Value).

C. In 32 bit system, store Value into SmallValue has 4 byte gain + preventing memory leak
in 64 bit system, store Value into SmallValue has 0 byte gain but this preventing memory leak

That's why we use Value on computing, SmallValue for storage

I am glad if you reduce Value<->SmallValue converting cost
